### ISSUE TITLE
Building the summator module from docs as shared module results in error

### ISSUE THREAD

**Issue by Athomield** [godotengine/godot#15675]
**Godot version:**

Master Branch from today (1/13/2018) 

**OS/device including version:**

Win 10 | MSVC 2017 x64

**Issue description:**

So after succeeding in building the summator module as a static library (by following the docs) I tried to do the same thing but make the library shared instead.
This appears to work at first but it generates a library with a name that is repeated twice **'summator.windows.tools.64.windows.tools.64.lib'** instead of **'summator.windows.tools.64.lib'.**

I've had this error before and I thought it was just from me or it will be fixed soon. But it is still persistent.

This is the full error .

`scons platform=windows summator_shared=yes
scons: Reading SConscript files ...
Detected MSVC compiler: amd64
Compiled program architecture will be a 64 bit executable (forcing bits=64).
Checking for C header file mntent.h... (cached) no
scons: done reading SConscript files.
scons: Building targets ...
[ 97%] [91mLinking Program        [95m==> [93mbin\godot.windows.tools.64.exe[0m
LINK : fatal error LNK1181: cannot open input file **'summator.windows.tools.64.windows.tools.64.lib'**
scons: *** [bin\godot.windows.tools.64.exe] Error 1181
scons: building terminated because of errors.`

**Steps to reproduce:**
Following the docs http://docs.godotengine.org/en/latest/development/cpp/custom_modules_in_cpp.html#creating-a-new-module using an MSVC 2017 compiler will reproduce this.

**Minimal reproduction project:**

**Comment by akien-mga** [comment]
Any Windows user to debug this and see if it's maybe just a documentation issue? (@vnen maybe? :))

**Comment by vnen** [comment]
Yes, this is a documentation problem because the doc author never considered MSVC when he wrote it.

The issue is that it's trying to link with `summator.windows.tools.64.windows.tools.64.lib`, which is a bogus name (part of it is repeated).

I think you can replace `env.Append(LIBS=[shared_lib_shim])` with simply `env.Append(LIBS=["summator"])`. Will try to test this later.

**Comment by akien-mga** [comment]
Retagging as documentation issue then, and definitely not critical for 3.0.

**Comment by vnen** [comment]
Eh, I don't know how that shared lib works at all. I essentially get the same as https://godotengine.org/qa/16495/getting-c-to-compile-as-a-shared-object-dll and it makes sense to act as such, since it references stuff that are inside the Godot source but are not added to the shared lib. I don't know enough to understand how this works with GCC.

**Comment by Athomield** [comment]
Is this issue very hard to solve ?

**Comment by logicor-cn** [comment]
I figured out how to get DLL built: https://github.com/godotengine/godot-docs/issues/2099.

@akien-mga It is more than just documentation, see my [comment](https://github.com/godotengine/godot-docs/issues/2099#issuecomment-458297994) in the other issue for details.

However, I got segfault when starting up the editor with dll built: 
https://github.com/godotengine/godot/issues/25448

**Comment by axsaucedo** [comment]
I can confirm this is still an issue in 3.2.3-stable

**Comment by mvccogo** [comment]
Still an issue in master. Is there any known fix for this? I need to access some low-level APIs from Godot such as the PhysicsServer that cannot be accessed by using a GDExtension, and recompiling the whole engine everytime is definitely not ideal.

**Comment by DanielSnd** [comment]
> Still an issue in master. Is there any known fix for this? I need to access some low-level APIs from Godot such as the PhysicsServer that cannot be accessed by using a GDExtension, and recompiling the whole engine everytime is definitely not ideal.

I'm also running into this issue on windows. I'm considering installing linux as a dual boot and using linux when developing the module I'm working on ._.

**Comment by SolventMercury** [comment]
Still having this issue trying to follow the documentation. Really hoping this sees a resolution soon, incredibly shoddy experience.

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot-docs/issues/5143.

Linking C++ modules as dynamic libraries isn't extensively tested since [godot-cpp](https://github.com/godotengine/godot-cpp) is now the main way to extend Godot with C++, so this will likely take a while to fix until a motivated contributor steps up.
